### ISSUE TITLE
Flutter crash report: NoSuchMethodError: The method 'shutdown' was called on null.

### ISSUE THREAD

**Issue by pulyaevskiy** [flutter/flutter#19362]
## command

flutter --no-color test --machine --plain-name happy path test test/happy_path_test.dart

## exception

NoSuchMethodError: NoSuchMethodError: The method 'shutdown' was called on null.
Receiver: null
Tried calling: shutdown()

```
#0      Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
#1      _Compiler.shutdown (package:flutter_tools/src/test/flutter_platform.dart:288)
<asynchronous suspension>
#2      _FlutterPlatform.close (package:flutter_tools/src/test/flutter_platform.dart:745)
<asynchronous suspension>
#3      Loader.close.<anonymous closure>.<anonymous closure> (package:test/src/runner/loader.dart:306)
<asynchronous suspension>
#4      MappedIterator.moveNext (dart:_internal/iterable.dart:392)
#5      Future.wait (dart:async/future.dart:385)
#6      Loader.close.<anonymous closure> (package:test/src/runner/loader.dart:304)
<asynchronous suspension>
#7      new Future.sync (dart:async/future.dart:222)
#8      AsyncMemoizer.runOnce (package:async/src/async_memoizer.dart:43)
#9      Loader.close (package:test/src/runner/loader.dart:302)
#10     Runner.close.<anonymous closure> (package:test/src/runner.dart:226)
<asynchronous suspension>
#11     new Future.sync (dart:async/future.dart:222)
#12     AsyncMemoizer.runOnce (package:async/src/async_memoizer.dart:43)
#13     Runner.close (package:test/src/runner.dart:198)
#14     main.close (package:test/src/executable.dart:165)
<asynchronous suspension>
#15     main (package:test/src/executable.dart:190)
<asynchronous suspension>
#16     runTests (package:flutter_tools/src/test/runner.dart:109)
<asynchronous suspension>
#17     TestCommand.runCommand (package:flutter_tools/src/commands/test.dart:216)
<asynchronous suspension>
#18     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:347)
#19     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/dart:async/async_patch.dart:77)
#20     _rootRunUnary (dart:async/zone.dart:1134)
#21     _CustomZone.runUnary (dart:async/zone.dart:1031)
#22     _FutureListener.handleValue (dart:async/future_impl.dart:129)
#23     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:638)
#24     Future._propagateToListeners (dart:async/future_impl.dart:667)
#25     Future._complete (dart:async/future_impl.dart:472)
#26     _SyncCompleter.complete (dart:async/future_impl.dart:51)
#27     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async-patch/dart:async/async_patch.dart:33)
#28     _rootRun (dart:async/zone.dart:1126)
#29     _CustomZone.run (dart:async/zone.dart:1023)
#30     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:949)
#31     _microtaskLoop (dart:async/schedule_microtask.dart:41)
#32     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#33     _runPendingImmediateCallback (dart:isolate-patch/dart:isolate/isolate_patch.dart:113)
#34     _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:166)
```

## flutter doctor

```
[✓] Flutter (Channel master, v0.5.6-pre.74, on Mac OS X 10.13.5 17F77, locale en-BY)
    • Flutter version 0.5.6-pre.74 at /Users/anatoly/flutter
    • Framework revision 3351423a42 (3 weeks ago), 2018-06-25 19:28:42 -0700
    • Engine revision 549c855e89
    • Dart version 2.0.0-dev.63.0.flutter-4c9689c1d2

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.3)
    • Android SDK at /Users/anatoly/Library/Android/sdk
    • Android NDK at /Users/anatoly/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 26.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.2

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 23.2.2
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.1.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 26.0.2
    • Dart plugin version 181.4892.1

[!] VS Code (version 1.24.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • iPhone 7 Plus • BC743601-D096-497F-8146-9F93F5EB3085 • ios • iOS 11.4 (simulator)

! Doctor found issues in 1 category.
```

**Comment by ThinkDigitalRepair** [comment]
Do you have any more error code above this? It usually tells you. This usually ends up not being an issue with Flutter, but rather an issue with your code. You ran some code on an object that was null;

**Comment by pulyaevskiy** [comment]
I found this report in a text file which Flutter generated without me noticing.

The stacktrace above does not look like originates from my project. It looks like there is an issue in `flutter_tools`.

**Comment by ThinkDigitalRepair** [comment]
Ah ok. I see it now

**Comment by zoechi** [comment]
I see this frequently as well.

**Comment by pulyaevskiy** [comment]
I haven't done any dev work in the past two weeks so it would be hard to tell.
I also don't know if there is a way to reproduce this error.

**Comment by zoechi** [comment]
I added the label "waiting for response" to get it closed in 4 weeks if you don't add a comment in the meantime.
We can revisit then and decide leaving it closed or reopen.

**Comment by zoechi** [comment]
Looks like a dup of #18610 which was fixed with #20553
